.github/workflows/tests.yml

```diff
@@ -0,0 +1,141 @@
+name: Tests
+
+on:
+  push:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  check_duplicate_runs:
```

Comment:
There is an annoying bug of Github Actions - when someone creates a new branch and pushes some commits here, and then creates a pull request, there are 2 copies of workflow running - one for a push event and another for created PR.

This step prevents the second copy of workflow to be run - if tests are already passed for push event then they will be not run twice while PR is created. Also they will not be run if source code hasn't been changed.

Reply:
This one is neat! 

Normally, I tend to avoid adding extra dependencies that have the full source code to be published at hand but this one is truly helpful!